### ISSUE TITLE
[Question] Programatically show the calendar?

### ISSUE THREAD

**Issue by jrmgx** [qodesmith/datepicker#44]
I did look for the doc about this but it does not seem to be possible,
still I'm asking: is it possible to programatically show the calendar?
Something like 

```
let cal = datepicker(".myclass");
cal.show();
```

The goal is to include the calendar in a more complex user flow than a simple click on an element...
Meanwhile I may just simulate a click to the bound class but it is not optimal.

Thank you for your great job!

**Comment by qodesmith** [comment]
@jrmgx At the moment there isn't, but I've been working on a new release and it will include both a show & hide method for calendar instances:

```javascript
const cal = datepicker('.myclass')

// Programmatically show the calendar. 
cal.show()

// Programmatically hide the calendar. 
cal.hide()
```

There will be a host of other API methods you'll be able to use. It's overdue. I've been finishing up the work with writing the tests, but I think I'll push a version today and finish up the tests after. I gotta get this out.

**Comment by jrmgx** [comment]
OMG this would be amazing,
I was going to fork and add it myself as the trick by using click event
isn't working so great.

Keep up the good work :)

**Comment by qodesmith** [comment]
FYI, you can see the progress I'm making in the [two-instances](https://github.com/qodesmith/datepicker/tree/two-instances) branch. That README has all the latest documentation (for this next release) minus the show & hide methods. I literally just added them now thanks to you! You can see the other API methods that are coming [here in the code](https://github.com/qodesmith/datepicker/blob/two-instances/datepicker.js#L878).

**Comment by jrmgx** [comment]
I've seen your changes here fc9a851bc7d718b3c0a9dd4cc485a27c37533152 and I am very grateful,
does it need some kind of build pass now? Because I don't see those new method in the min version of the lib...

**Comment by qodesmith** [comment]
I haven't build the minified version, so you're correct. And this is still all on the two-instances branch. I'm going to merge to master today and just deal with writing all the rest of the tests later.

**Comment by jrmgx** [comment]
Ok, no rush, I will build it myself and do my dev with this version, I can wait a few days...
You're already did way too much for today :p

**Comment by qodesmith** [comment]
Just pushed a built minified version. You can use that.

**Comment by jrmgx** [comment]
Thanks to you I was able to include the datepicker into an existing date-time picking flow
see 
![nov-05-2018 13-59-44](https://user-images.githubusercontent.com/42989/48029623-7d693b00-e103-11e8-85fc-9ecb571843d4.gif)
With the new v4 you just pushed, I guess this issue is worth closing now.

**Comment by qodesmith** [comment]
@jrmgx Dude, that looks great!

**Comment by open-environment** [comment]
any example code on this? I'm trying to get this to work but coming up short. 

HTML
`                                    <div class="input-group date" data-provide="datepicker">
                                        <input autocomplete="off" class="form-control" data-val="true" id="Assessment_ASSESSMENT_DT" type="text">
                                        <div class="input-group-addon">
                                            <span class="glyphicon glyphicon-calendar"></span>
                                        </div>
                                    </div>`

JS:
`            const picker = datepicker('#Assessment_ASSESSMENT_DT', {
                formatter: (input, date, instance) => {
                    const value = date.toLocaleDateString()
                    input.value = value // => '1/1/2099'
                }
            });

            $(".input-group-addon").click(function () {
                picker.show();
            });`

**Comment by sfusato** [comment]
I have a similar usecase as @jrmgx (by selecting a specific option from a drop-down, the calendar is shown).

What I tried:
1. using the `show` method => I could see the `top` and `left` css properties appear in the `style` attribute of the calendar and the `qs-hidden` class highlighted by Chrome, but not removed. So, the calendar wasn't showing;
2. manually removing the `qs-hidden` class => class highlighted by Chrome, which means something added it back right after I attempted to remove it as before;

I think there's a race condition here: when I click that specific option, I trigger a `click` event, to which DatePicker automatically reacts (after my code) and basically hides the calendar:
```
 if (type === 'click') {
    // Anywhere other than the calendar - close the calendar.
    if (!instance) return datepickers.forEach(hideCal)
```
https://github.com/qodesmith/datepicker/blob/two-instances/datepicker.js#L801

For now I use a `setTimeout` to call `show` 10 ms later. This is an ugly hack. Wonder if there's something better?

@open-environment perhaps you ran into the same race condition. From the code you shared it certainly looks like so.
